CoreFoundation/Base.subproj/CoreFoundation_Prefix.h

```diff
@@ -223,6 +228,32 @@ CF_INLINE size_t malloc_size(void *memblock) {
 
 #endif
 
+#if DEPLOYMENT_TARGET_FREEBSD
+#define HAVE_STRUCT_TIMESPEC 1
+
+#define CF_PRIVATE __attribute__((visibility("hidden")))
+#define __strong
+#define __weak
+
+// Implemented in CFPlatform.c
+bool OSAtomicCompareAndSwapPtr(void *oldp, void *newp, void *volatile *dst);
+bool OSAtomicCompareAndSwapLong(long oldl, long newl, long volatile *dst);
+bool OSAtomicCompareAndSwapPtrBarrier(void *oldp, void *newp, void *volatile *dst);
+bool OSAtomicCompareAndSwap64Barrier( int64_t __oldValue, int64_t __newValue, volatile int64_t *__theValue );
+
+int32_t OSAtomicDecrement32Barrier(volatile int32_t *dst);
+int32_t OSAtomicIncrement32Barrier(volatile int32_t *dst);
+int32_t OSAtomicIncrement32(volatile int32_t *theValue);
+int32_t OSAtomicDecrement32(volatile int32_t *theValue);
+
+int32_t OSAtomicAdd32( int32_t theAmount, volatile int32_t *theValue );
+int32_t OSAtomicAdd32Barrier( int32_t theAmount, volatile int32_t *theValue );
+bool OSAtomicCompareAndSwap32Barrier( int32_t oldValue, int32_t newValue, volatile int32_t *theValue );
+
+void OSMemoryBarrier();
+
```

Comment:
does FreeBSD have malloc_size? I was under the impression that it was similar to linux in that it defined malloc_usable_size instead.